### ISSUE TITLE
OpenTofu: checksum list has unexpected SHA-256 hash

### ISSUE THREAD

**Issue by nikstuckenbrock** [gridscale/terraform-provider-gridscale#399]
### Terraform Version (OpenTofu Version)
OpenTofu v1.8.2 on linux_amd64

### Affected Resource(s)
General issue not affecting special resources.

### Terraform Configuration Files
provider.tf
```tofu
terraform {
  required_providers {
    gridscale = {
      source = "gridscale/gridscale"
    }
  }
}

provider "gridscale" {
  uuid  = var.gridscale_uuid
  token = var.gridscale_token
}
```

### Debug Output
```

Initializing provider plugins...
- Finding latest version of hashicorp/template...
2024-10-28T16:17:35.529+0100 [DEBUG] Service discovery for registry.opentofu.org at https://registry.opentofu.org/.well-known/terraform.json
2024-10-28T16:17:35.746+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/hashicorp/template/versions
- Finding latest version of gridscale/gridscale...
2024-10-28T16:17:35.875+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/gridscale/gridscale/versions
- Finding latest version of hashicorp/local...
2024-10-28T16:17:35.995+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/hashicorp/local/versions
2024-10-28T16:17:36.105+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/hashicorp/template/2.2.0/download/linux/amd64
2024-10-28T16:17:36.202+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-template/releases/download/v2.2.0/terraform-provider-template_2.2.0_SHA256SUMS
2024-10-28T16:17:36.964+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-template/releases/download/v2.2.0/terraform-provider-template_2.2.0_SHA256SUMS.sig
- Installing hashicorp/template v2.2.0...
2024-10-28T16:17:37.513+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-template/releases/download/v2.2.0/terraform-provider-template_2.2.0_linux_amd64.zip
2024-10-28T16:17:38.844+0100 [DEBUG] Validating GPG signature of provider package /tmp/terraform-provider1772639085
2024-10-28T16:17:38.852+0100 [DEBUG] Provider signed by 0C0AF313E5FD9F80 OpenTofu (This key is used to sign opentofu providers) <core@opentofu.org>
- Installed hashicorp/template v2.2.0 (signed, key ID 0C0AF313E5FD9F80)
2024-10-28T16:17:39.309+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/gridscale/gridscale/1.27.0/download/linux/amd64
2024-10-28T16:17:39.469+0100 [DEBUG] GET https://github.com/gridscale/terraform-provider-gridscale/releases/download/v1.27.0/terraform-provider-gridscale_1.27.0_SHA256SUMS
2024-10-28T16:17:40.181+0100 [DEBUG] GET https://github.com/gridscale/terraform-provider-gridscale/releases/download/v1.27.0/terraform-provider-gridscale_1.27.0_SHA256SUMS.sig
- Installing gridscale/gridscale v1.27.0...
2024-10-28T16:17:40.731+0100 [DEBUG] GET https://github.com/gridscale/terraform-provider-gridscale/releases/download/v1.27.0/terraform-provider-gridscale_1.27.0_linux_amd64.zip
2024-10-28T16:17:41.839+0100 [DEBUG] GET https://registry.opentofu.org/v1/providers/hashicorp/local/2.5.2/download/linux/amd64
2024-10-28T16:17:42.030+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-local/releases/download/v2.5.2/terraform-provider-local_2.5.2_SHA256SUMS
2024-10-28T16:17:42.839+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-local/releases/download/v2.5.2/terraform-provider-local_2.5.2_SHA256SUMS.sig
- Installing hashicorp/local v2.5.2...
2024-10-28T16:17:43.399+0100 [DEBUG] GET https://github.com/opentofu/terraform-provider-local/releases/download/v2.5.2/terraform-provider-local_2.5.2_linux_amd64.zip
2024-10-28T16:17:44.596+0100 [DEBUG] Validating GPG signature of provider package /tmp/terraform-provider2608281752
2024-10-28T16:17:44.605+0100 [DEBUG] Provider signed by 0C0AF313E5FD9F80 OpenTofu (This key is used to sign opentofu providers) <core@opentofu.org>
- Installed hashicorp/local v2.5.2 (signed, key ID 0C0AF313E5FD9F80)

Providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://opentofu.org/docs/cli/plugins/signing/
╷
│ Error: Failed to install provider
│ 
│ Error while installing gridscale/gridscale v1.27.0: checksum list has unexpected SHA-256 hash 03fe0964cf7e7d06e86bb40573bd69dbb67ce9ae9c69dd71774876a422fbf36a (expected
│ 8bbf514982e0c9cf52f7250829f9e04ce6a56fa913d885ff1b617da60a4bca4a)
```

### Expected Behavior
Successfull installation.

### Actual Behavior
Error when installing matching the hashes.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `tofu init`

**Comment by nikstuckenbrock** [comment]
Works for older versions. Only the latest version v1.27.0 has problems. v1.26.0 works fine.

**Comment by nvthongswansea** [comment]
@nikstuckenbrock Hi, we've recently updated our GPG key. I've opened a new [PR](https://github.com/opentofu/registry/issues/1223) in opentofy registry to update gridscale GPG key.

**Comment by nikstuckenbrock** [comment]
The PRs [#1223](https://github.com/opentofu/registry/issues/1223) and [#1225](https://github.com/opentofu/registry/pull/1225) don't fix the problem.

**Comment by nikstuckenbrock** [comment]
Now version 1.26.0 isn't working anymore too.

**Comment by nvthongswansea** [comment]
@nikstuckenbrock In that case, I will need to contact opentofu team regarding this issue. Because it is working ok with terraform.

**Comment by nikstuckenbrock** [comment]
@nvthongswansea There seems to be a version diff between the OpenTofu and Terraform registry:

- https://search.opentofu.org/provider/opentofu/gridscale/latest
- https://registry.terraform.io/providers/gridscale/gridscale/latest

At least that is what i would assume looking at the docs.

But it sees to be registered in some way:
- https://github.com/opentofu/registry/blob/main/providers/g/gridscale/gridscale.json

**Comment by nvthongswansea** [comment]
@nikstuckenbrock I've opened a PR [here](https://github.com/opentofu/registry/pull/1249).

**Comment by ghost** [comment]
Hey folks, I just merged the PR which should cause version 1.27.0 to be re-indexed. I'm happy to assist either here or on the OpenTofu Slack should you need it.

**Comment by nikstuckenbrock** [comment]
Thanks @abstractionfactory @nvthongswansea ,

issue seems to be fixed.
